### PR TITLE
[Feature] Support aliyun cloud credential

### PR DESCRIPTION
**Pull request by miomiocat**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/credential/aws/AWSCloudConfiguration.java [hunk 1]
```diff
@@ -54,7 +54,7 @@ public void applyToConfiguration(Configuration configuration) {
         configuration.set("fs.s3.impl", "org.apache.hadoop.fs.s3a.S3AFileSystem");
         configuration.set("fs.s3a.impl", "org.apache.hadoop.fs.s3a.S3AFileSystem");
         // Below storage using s3 compatible storage api
-        configuration.set("fs.oss.impl", "org.apache.hadoop.fs.s3a.S3AFileSystem");
+        configuration.set("fs.ks3.impl", "org.apache.hadoop.fs.s3a.S3AFileSystem");
```
[the remark this answers is not in the document]
Comment (miomiocat) [reply fe/fe-core/src/main/java/com/starrocks/credential/aws/AWSCloudConfiguration.java]: done

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 20 / 47 (42.55%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/credential/aliyun/AliyunCloudConfiguration.java|0|14|00.00%|[33, 34, 35, 36, 40, 42, 43, 44, 45, 49, 50, 54, 60, 65]|
|:large_blue_circle:|com/starrocks/credential/aliyun/AliyunCloudCredential.java|9|21|42.86%|[42, 43, 44, 45, 46, 47, 56, 57, 58, 59, 63, 73]|
|:large_blue_circle:|com/starrocks/credential/aliyun/AliyunCloudConfigurationFactory.java|7|8|87.50%|[46]|
|:large_blue_circle:|com/starrocks/credential/CloudType.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/credential/CloudCredentialUtil.java|2|2|100.00%|[]|

File: gensrc/thrift/CloudConfiguration.thrift [hunk 2]
```diff
@@ -19,7 +19,8 @@ enum TCloudType {
     DEFAULT,
     AWS,
     AZURE,
-    GCP
+    GCP,
+    OSS
```
Comment (imay) [inline gensrc/thrift/CloudConfiguration.thrift:23]: I think you need to use AliCloud rather than OSS, because others are cloud service vender names.

Comment (Smith-Cruise) [reply gensrc/thrift/CloudConfiguration.thrift:23]: +1

File: fe/fe-core/src/main/java/com/starrocks/credential/aliyun/AliyunCloudConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    private final AliyunCloudCredential aliyunCloudCredential;
+
+    public AliyunCloudConfiguration(AliyunCloudCredential aliyunCloudCredential) {
+        Preconditions.checkNotNull(aliyunCloudCredential);
+        this.aliyunCloudCredential = aliyunCloudCredential;
+    }
+
+    @Override
+    public void toThrift(TCloudConfiguration tCloudConfiguration) {
+        tCloudConfiguration.setCloud_type(TCloudType.OSS);
```
Comment (Smith-Cruise) [inline fe/fe-core/src/main/java/com/starrocks/credential/aliyun/AliyunCloudConfiguration.java:40]: just using aws's logic to send thrift to be

File: fe/fe-core/src/main/java/com/starrocks/credential/aliyun/AliyunCloudConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+import com.starrocks.thrift.TCloudType;
+import org.apache.hadoop.conf.Configuration;
+
+import java.util.LinkedList;
+import java.util.List;
+
+public class AliyunCloudConfiguration implements CloudConfiguration {
+
+    private final AliyunCloudCredential aliyunCloudCredential;
+
```
Comment (Smith-Cruise) [inline fe/fe-core/src/main/java/com/starrocks/credential/aliyun/AliyunCloudConfiguration.java:32]: maybe add an `enable_ssl` option, you can refer it to AWSCloudConfiguration

File: be/src/fs/credential/cloud_configuration_factory.h [hunk 5]
```diff
@@ -80,6 +84,23 @@ class CloudConfigurationFactory {
         return aws_cloud_configuration;
     }
 
+    static const AliyunCloudConfiguration create_aliyun(const TCloudConfiguration& t_cloud_configuration) {
```
Comment (Smith-Cruise) [inline be/src/fs/credential/cloud_configuration_factory.h:87]: add some comment, like `// used for aliyun emr's StarRocks` ?

File: gensrc/thrift/CloudConfiguration.thrift (hunk 2 above)
Comment (miomiocat) [reply gensrc/thrift/CloudConfiguration.thrift]: done

File: fe/fe-core/src/main/java/com/starrocks/credential/CloudConfigurationConstants.java [hunk 6]
```diff
@@ -93,4 +93,9 @@ public class CloudConfigurationConstants {
     public static final String HDFS_KERBEROS_PRINCIPAL = "kerberos_principal";
     public static final String HDFS_KERBEROS_KEYTAB = "kerberos_keytab";
     public static final String HDFS_KERBEROS_KEYTAB_CONTENT = "kerberos_keytab_content";
+
+    // Credential for Aliyun OSS
+    public static final String OSS_ACCESS_KEY = "aliyun.oss.access_key_id";
```
[the remark this answers is not in the document]
Comment (miomiocat) [reply fe/fe-core/src/main/java/com/starrocks/credential/CloudConfigurationConstants.java]: done

File: fe/fe-core/src/main/java/com/starrocks/credential/CloudConfigurationConstants.java [hunk 7]
```diff
@@ -93,4 +93,9 @@ public class CloudConfigurationConstants {
...
+    public static final String OSS_ACCESS_SECRET = "aliyun.oss.access_key_secret";
```
[the remark this answers is not in the document]
Comment (miomiocat) [reply fe/fe-core/src/main/java/com/starrocks/credential/CloudConfigurationConstants.java]: done

File: fe/fe-core/src/main/java/com/starrocks/credential/aliyun/AliyunCloudConfiguration.java (hunk 3 above)
Comment (miomiocat) [reply fe/fe-core/src/main/java/com/starrocks/credential/aliyun/AliyunCloudConfiguration.java]: done

File: fe/fe-core/src/main/java/com/starrocks/credential/aliyun/AliyunCloudConfiguration.java (hunk 4 above)
Comment (miomiocat) [reply fe/fe-core/src/main/java/com/starrocks/credential/aliyun/AliyunCloudConfiguration.java:32]: keep default here

File: be/src/fs/credential/cloud_configuration_factory.h (hunk 5 above)
Comment (miomiocat) [reply be/src/fs/credential/cloud_configuration_factory.h:88]: done

Comment (miomiocat) [reply be/src/fs/credential/cloud_configuration_factory.h:88]: done

File: fe/fe-core/src/main/java/com/starrocks/credential/aliyun/AliyunCloudConfiguration.java [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+    public AliyunCloudConfiguration(AliyunCloudCredential aliyunCloudCredential) {
+        Preconditions.checkNotNull(aliyunCloudCredential);
+        this.aliyunCloudCredential = aliyunCloudCredential;
+    }
+
+    @Override
+    public void toThrift(TCloudConfiguration tCloudConfiguration) {
+        tCloudConfiguration.setCloud_type(TCloudType.AWS);
+
+        List<TCloudProperty> properties = new LinkedList<>();
```
Comment (Smith-Cruise) [inline fe/fe-core/src/main/java/com/starrocks/credential/aliyun/AliyunCloudConfiguration.java:42]: `properties.add(new TCloudProperty(CloudConfigurationConstants.AWS_S3_ENABLE_SSL, String.valueOf(true)));`

Add this, recommend to enable SSL by default when execute in be

File: fe/fe-core/src/main/java/com/starrocks/credential/aliyun/AliyunCloudConfiguration.java [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+    private final AliyunCloudCredential aliyunCloudCredential;
+
+    public AliyunCloudConfiguration(AliyunCloudCredential aliyunCloudCredential) {
+        Preconditions.checkNotNull(aliyunCloudCredential);
+        this.aliyunCloudCredential = aliyunCloudCredential;
+    }
+
+    @Override
+    public void toThrift(TCloudConfiguration tCloudConfiguration) {
+        tCloudConfiguration.setCloud_type(TCloudType.AWS);
```
Comment (Smith-Cruise) [inline fe/fe-core/src/main/java/com/starrocks/credential/aliyun/AliyunCloudConfiguration.java:40]: Add comment, reuse be's aws logic

File: fe/fe-core/src/main/java/com/starrocks/credential/aliyun/AliyunCloudConfiguration.java [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    @Override
+    public void applyToConfiguration(Configuration configuration) {
+        aliyunCloudCredential.applyToConfiguration(configuration);
+    }
+
+    @Override
+    public CloudType getCloudType() {
+        return CloudType.OSS;
```
Comment (Smith-Cruise) [inline fe/fe-core/src/main/java/com/starrocks/credential/aliyun/AliyunCloudConfiguration.java:54]: CloudType is cloud vendor name, you should use aliyun instead.

File: fe/fe-core/src/main/java/com/starrocks/credential/aliyun/AliyunCloudCredential.java [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+import com.starrocks.credential.CloudConfigurationConstants;
+import com.starrocks.credential.CloudCredential;
+import com.starrocks.thrift.TCloudProperty;
+import org.apache.hadoop.conf.Configuration;
+
+import java.util.List;
+
+public class AliyunCloudCredential implements CloudCredential {
+
+    private final String accessKeyId;
```
Comment (Smith-Cruise) [inline fe/fe-core/src/main/java/com/starrocks/credential/aliyun/AliyunCloudCredential.java:28]: accessKey

File: fe/fe-core/src/main/java/com/starrocks/credential/aliyun/AliyunCloudCredential.java [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+    private final String accessKeySecret;
```
Comment (Smith-Cruise) [inline fe/fe-core/src/main/java/com/starrocks/credential/aliyun/AliyunCloudCredential.java:29]: secreyKey

File: fe/fe-core/src/main/java/com/starrocks/credential/aliyun/AliyunCloudConfiguration.java (hunk 4 above)
Comment (Smith-Cruise) [reply fe/fe-core/src/main/java/com/starrocks/credential/aliyun/AliyunCloudConfiguration.java:32]: ok

File: be/src/fs/credential/cloud_configuration_factory.h [hunk 13]
```diff
@@ -49,6 +49,10 @@ static const std::string AWS_S3_ENABLE_PATH_STYLE_ACCESS = "aws.s3.enable_path_s
      */
 static const std::string AWS_S3_ENABLE_SSL = "aws.s3.enable_ssl";
 
+static const std::string ALIYUN_OSS_ACCESS_KEY = "aliyun.oss.access_key_id";
```
Comment (Smith-Cruise) [inline be/src/fs/credential/cloud_configuration_factory.h:52]: access_key

File: be/src/fs/credential/cloud_configuration_factory.h [hunk 14]
```diff
@@ -49,6 +49,10 @@ static const std::string AWS_S3_ENABLE_PATH_STYLE_ACCESS = "aws.s3.enable_path_s
...
+static const std::string ALIYUN_OSS_SECRET_KEY = "aliyun.oss.access_key_secret";
```
Comment (Smith-Cruise) [inline be/src/fs/credential/cloud_configuration_factory.h:53]: secret_key

File: be/src/fs/credential/cloud_configuration_factory.h (hunk 13 above)
Comment (stephen-shelby) [inline be/src/fs/credential/cloud_configuration_factory.h:52]: keep with aliyun.oss.access_key ?

**Review by Smith-Cruise — APPROVED** [review]

**Review by stephen-shelby — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/credential/CloudConfigurationConstants.java [hunk 15]
```diff
@@ -93,4 +93,9 @@ public class CloudConfigurationConstants {
     public static final String HDFS_KERBEROS_PRINCIPAL = "kerberos_principal";
     public static final String HDFS_KERBEROS_KEYTAB = "kerberos_keytab";
     public static final String HDFS_KERBEROS_KEYTAB_CONTENT = "kerberos_keytab_content";
+
+    // Credential for Aliyun OSS
+    public static final String ALIYUN_OSS_ACCESS_KEY = "aliyun.oss.access_key";
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/credential/CloudConfigurationConstants.java:98]: Why is it different from ALIYUN_OSS_ACCESS_KEY?

**Comment by imay** [comment]
@miomiocat 
You need to update documents according to this change list.

**Pull request merged**